tools/ci_build/build.py

```diff
@@ -378,23 +373,6 @@ def generate_build_tree(cmake_path, source_dir, build_dir, cuda_home, cudnn_home
         cmake_args += ["-DCMAKE_TOOLCHAIN_FILE=" + args.android_ndk_path + "/build/cmake/android.toolchain.cmake",
                 "-DANDROID_PLATFORM=android-" + str(args.android_api),
                 "-DANDROID_ABI=" + str(args.android_abi)]
-    
-    if args.ios:
-        needed_args = [len(args.ios_sysroot) == 0, not args.arm64 and not args.arm, len(args.ios_toolchain_dir) == 0]
-        arg_names = ["--ios_sysroot <path to sysroot>", "--arm or --arm64", "--ios_toolchain_dir <path to toolchain>"]
-        if any(needed_args):
-            raise BuildError("iOS build canceled due to missing arguments: " + (val + "," for val, cond in zip(arg_names, needed_args) if cond))
```

Comment:
I don't think this works, you're concatenating string and generator

Reply:
Tested `"iOS build canceled due to missing arguments: " + (val + "," for val, cond in zip(['a', 'b'], [True, False]) if cond)` in an interactive shell and it fails indeed :)

Reply:
Oh just realized you're removing this code instead of adding it. Never mind then